yesod-test/test/main.hs

```diff
@@ -776,6 +814,48 @@ getIntegerR = do
     app <- getYesod
     pure $ T.pack $ show (routedAppInteger app)
 
+tupleField = Field parse view UrlEncoded
+  where
+  parse :: [Text] -> [FileInfo] -> (HandlerFor RoutedApp) (Either (SomeMessage (HandlerSite (HandlerFor RoutedApp))) (Maybe (Int, Text)))
+  parse rawVals@[someNumber, someText] _fileVals =  
+    case readMaybe (T.unpack someNumber)::Maybe Int of
+      Nothing -> pure $ Left "The first element is not Int"
+      Just n -> pure $ Right $ Just (n, someText)
+  parse rawVals _ = pure $ Left "The number of elements in the first argument of the parse is not equal to two"
+
+  view theId name attrs _eVal isReq =
+    [whamlet|
+      <input name=#{name}>
+      <input name=#{name}>
+    |]
+
+tupleForm :: Html -> MForm Handler (FormResult (Int, Text), Widget)
+tupleForm = renderDivs $ areq tupleField "" { fsName = Just "data" } Nothing
+
+getTupleR :: Handler Html
+getTupleR = do
+    ((result, widget), enctype) <- runFormPost tupleForm
```

Comment:
The result of running this form isn't used in this function, so I would expect this to generate a warning. If the compiler is generating warnings, we should writer neater code.

```suggestion
    ((_, widget), enctype) <- runFormPost tupleForm
```